### PR TITLE
Add repeat function

### PR DESCRIPTION
**Pull request by highker**
Allow repeating an element for a given number of times. The return
result is an array.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java [hunk 1]
```diff
@@ -933,6 +933,28 @@ public void testArrayRemove()
     }
 
     @Test
+    public void testRepeat()
+            throws Exception
+    {
+        // concrete values
+        assertFunction("REPEAT(1, 5)", new ArrayType(INTEGER), ImmutableList.of(1, 1, 1, 1, 1));
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java]: Test zero count.

Comment (highker) [inline presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java]: resolved

**Comment by haozhun** [comment]
The implementation looks good. But let's wait for your limit investigation.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/SequenceFunction.java [hunk 2]
```diff
@@ -82,6 +83,7 @@ public static Block sequenceTimestampYearToMonth(
                 "sequence end value should be greater than or equal to start value if step is greater than zero otherwise end should be less than start");
 
         int length = toIntExact(diffTimestamp(session, MONTH, start, end) / step + 1);
+        checkCondition(length <= MAX_LENGTH, INVALID_FUNCTION_ARGUMENT, "sequence should be no greater than " + MAX_LENGTH);
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/SequenceFunction.java]: Result of sequence function is capped at MAX_LENGTH entries

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.util.Failures.checkCondition;
+import static java.lang.Math.toIntExact;
+
+@ScalarFunction("repeat")
+@Description("Repeat an element for a given number of times")
+public final class RepeatFunction
+{
+    private static final long MAX_LENGTH = 10_000;
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java]: Rename this to `MAX_RESULT_ENTRIES`

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        for (int i = 0; i < count; i++) {
+            blockBuilder.appendNull();
+        }
+        return blockBuilder.build();
+    }
+
+    private static void checkMaxSize(long bytes, long count)
+    {
+        checkCondition(bytes <= (MAX_SIZE_IN_BYTES + count) / count,
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java]: first argument on its own line

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java (hunk 3 above)
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java]: `MAX_RESULT_ENTRIES`

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        }
+        for (int i = 0; i < count; i++) {
+            type.writeDouble(blockBuilder, element);
+        }
+        return blockBuilder.build();
+    }
+
+    private static BlockBuilder createBlockBuilder(Type type, long count)
+    {
+        checkCondition(count <= MAX_LENGTH, INVALID_FUNCTION_ARGUMENT, "repeat count should be no greater than " + MAX_LENGTH);
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java]: `count argument of repeat function must be less than or equal to MAX_LENGTH`

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        checkCondition(count >= 0, INVALID_FUNCTION_ARGUMENT, "repeat count must be no smaller than zero");
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java]: count argument of repeat function must be greater than or equal to 0

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+            blockBuilder.appendNull();
+        }
+        return blockBuilder.build();
+    }
+
+    private static void checkMaxSize(long bytes, long count)
+    {
+        checkCondition(bytes <= (MAX_SIZE_IN_BYTES + count) / count,
+                INVALID_FUNCTION_ARGUMENT,
+                "repeat should not create an array greater than %s bytes",
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/RepeatFunction.java]: `result of repeat function must not take more than MAX_SIZE_IN_BYTES bytes`

For coherency, don't use %s.

File: presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java [hunk 8]
```diff
@@ -937,6 +937,39 @@ public void testArrayRemove()
...
+
+        // 0 counts
+        assertFunction("REPEAT(cast(null as bigint), 0)", new ArrayType(BIGINT), ImmutableList.of());
+        assertFunction("REPEAT(1, 0)", new ArrayType(INTEGER), ImmutableList.of());
+        assertFunction("REPEAT('varchar', 0)", new ArrayType(createVarcharType(7)), ImmutableList.of());
+        assertFunction("REPEAT(true, 0)", new ArrayType(BOOLEAN), ImmutableList.of());
+        assertFunction("REPEAT(0.5, 0)", new ArrayType(DOUBLE), ImmutableList.of());
+        assertFunction("REPEAT(array[1], 0)", new ArrayType(new ArrayType(INTEGER)), ImmutableList.of());
+
+        // failure modes
```
Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java]: illegal inputs

File: presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java [hunk 9]
```diff
@@ -937,6 +937,39 @@ public void testArrayRemove()
...
+    {
+        // concrete values
+        assertFunction("REPEAT(1, 5)", new ArrayType(INTEGER), ImmutableList.of(1, 1, 1, 1, 1));
+        assertFunction("REPEAT('varchar', 3)", new ArrayType(createVarcharType(7)), ImmutableList.of("varchar", "varchar", "varchar"));
+        assertFunction("REPEAT(true, 1)", new ArrayType(BOOLEAN), ImmutableList.of(true));
+        assertFunction("REPEAT(0.5, 4)", new ArrayType(DOUBLE), ImmutableList.of(0.5, 0.5, 0.5, 0.5));
+        assertFunction("REPEAT(array[1], 4)", new ArrayType(new ArrayType(INTEGER)), ImmutableList.of(ImmutableList.of(1), ImmutableList.of(1), ImmutableList.of(1), ImmutableList.of(1)));
+
+        // null values
+        assertFunction("REPEAT(cast(null as bigint), 4)", new ArrayType(BIGINT), asList(null, null, null, null));
```
Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java]: Test `null` without cast

**Pull request merged**